plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/callables/PartiallySpecialKotlinFunction.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2010-2024 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.embeddings.callables
+
+import org.jetbrains.kotlin.formver.conversion.StmtConversionContext
+import org.jetbrains.kotlin.formver.embeddings.expression.ExpEmbedding
+import org.jetbrains.kotlin.formver.embeddings.expression.OperatorExpEmbeddings.AddStringString
+import org.jetbrains.kotlin.formver.embeddings.types.buildFunctionPretype
+import org.jetbrains.kotlin.formver.embeddings.types.equalToType
+import org.jetbrains.kotlin.formver.embeddings.types.nullableAny
+import org.jetbrains.kotlin.formver.viper.MangledName
+import org.jetbrains.kotlin.formver.viper.ast.Method
+
+/**
+ * Functions that should sometimes be handled separately by our conversion depending on arguments they're called with.
+ */
+interface PartiallySpecialKotlinFunction : SeparatelyHandledKotlinFunction {
```

Comment:
I don't quite understand the way this class is meant to be used.  Do I understand correctly that `baseEmbedding` is the function that is called if no special case is triggered, and `tryInsertCall` is what decides whether the special behaviour should be used or not?  Is this all being introduced only for the sake of `StringPlusAnyFunction`, or are there other cases?